<a href="https://colab.research.google.com/github/Gise08/Aprendizaje-Autom-tico/blob/main/Clase4/AA_Clase4_2.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

In [2]:
import pandas as pd
import numpy as np
import matplotlib.pyplot as plt
from sklearn import linear_model
from sklearn.metrics import accuracy_score, confusion_matrix

# Cargamos el dataset propuesto para la actividad

df_logs = pd.read_csv('usuarios_win_mac_lin.csv')

# Visualizamos las características principales:
# Duración, Páginas, Acciones y Valor de acciones
df_logs.head()

,duracion,paginas,acciones,valor,clase
0,7.0,2,4,8,2
1,21.0,2,6,6,2
2,57.0,2,4,4,2
3,101.0,3,6,12,2
4,109.0,2,6,12,2


Realizamos el análisis exploratorio de datos para verificar la calidad y precisión. Nuestra información de entrada consiste en 4 características: duración de la visita, páginas vistas, cantidad de acciones y valor de las acciones

In [3]:
# Definimos X e Y
X = np.array(df_logs.drop(['clase'], axis=1))
y = np.array(df_logs['clase'])

# Creamos el objeto clasificador de Regresión Logística

clasificador = linear_model.LogisticRegression(solver='lbfgs', C=100, max_iter=1000)

# Ajustamos el modelo a nuestros datos (Aprendizaje Supervisado)
clasificador.fit(X, y)

LogisticRegression(C=100, max_iter=1000)

Implementamos un modelo de Regresión Logística. Mediante el método fit(), el algoritmo aprende a distinguir las etiquetas basándose en el comportamiento del usuario en la web.

In [4]:
# Realizamos la predicción sobre los mismos datos para ver el ajuste inicial
predicciones = clasificador.predict(X)

# Calculamos la precisión (Accuracy)
precision = accuracy_score(y, predicciones)
print(f"Precisión del modelo: {round(precision * 100, 2)}%")

# Ejemplo de predicción para un nuevo usuario:
# [Duración, Páginas, Acciones, Valor]
nuevo_usuario = np.array([[60, 5, 10, 50]])
print(f"Predicción para el nuevo usuario: {clasificador.predict(nuevo_usuario)}")

Precisión del modelo: 78.24%
Predicción para el nuevo usuario: [1]


Evaluamos el modelo utilizando la métrica de precisión (accuracy_score), que nos indica qué tan bien predice las clases correctas. La visualización de estas predicciones nos permite tomar decisiones informadas sobre la efectividad del modelo en aplicaciones reales, como el análisis financiero o el comportamiento web.Un 78.24% de precisión (accuracy) es un valor sólido, esto nos indica que el modelo de Regresión Logística clasificó correctamente a casi 8 de cada 10 usuarios en el sistema operativo que realmente utilizaban. El resultado [1] significa que, el modelo determinó que el perfil de comportamiento corresponde a un usuario de Macintosh.